### DISCUSSION TITLE
Suggesting a pagination feature to enable users know how many pages are in a particular lesson

### DISCUSSION THREAD

**Discussion by Ronnie-Mercy** [oppia/oppia-android#4646] [General Discussion]
I'd like to share an idea I came up with while using the Oppia app recently. Simple pagination below the continue/submit button will help the user know how many more lessons they have left and also a clear indication of what page they are on currently. I added the screen I'm talking about and also
![image](https://user-images.githubusercontent.com/115509229/195346217-58d793a3-d30a-4080-9060-2235cdb5e05c.png)
 a simple mockup below that can be improved.

**Comment by BenHenning** [comment]
Thanks for filing this @Ronnie-Mercy. I've checked and we don't yet have an issue tracking this, so this is a good place to capture the context.

For some background: we do have plans to introduce this functionality, but it'll probably be on the order of quarters before we get to it. This problem is actually complex since the lesson can have a changing number of cards depending on a particular learner's playthrough, and can even involve navigating back to an earlier part of the lesson. On the web side, we're working on introducing something called "full checkpointing" which allows lesson creators to specify specific _points_ in the lesson that are considered "completions" that we can also use to show progress. Until that work is completed, this will be on hold unless we hear other feedback that this is a major blocker. Either way, this is important for us to address and is definitely on our radar.

Edit: https://github.com/oppia/oppia-android/issues/46521 has been filed to track this.

**Comment by Ronnie-Mercy** [comment]
Hello @BenHenning thank you very much for this detailed explanation of how your team will be addressing this issue!  Also about the "full checkpointing" feature, and my suggestion being on your radar, what would it entail for me to work on contributing further to this? like is there any way I can get to further my work on this?

**Reply by seanlip** [reply]
Hi @Ronnie-Mercy, 

Are you an Android developer? If so, you can get started with helping out with the codebase by following [these instructions](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android). If not, you'll probably need to wait for the technical team to look into it.

**Reply by Ronnie-Mercy** [reply]
Hi, @seanlip  I'm not an android developer but a UI/UX Designer. I'll wait for the technical team to look into it! thank you!

**Reply by BenHenning** [reply]
@Ronnie-Mercy do you maybe have UI/UX design experience with Android? We also have contribution opportunities outside of technical work (see https://github.com/oppia/oppia/wiki/Contributing-to-Oppia%27s-design) if you're interested in helping out. :) (Note that we have opportunities for both Android & web UI/UX design, but we could always use more help on Android).

**Reply by Ronnie-Mercy** [reply]
Yes, I have good experience designing for Android and am very interested in helping out! I'm an outreachy intern
